superbench/common/utils/file_handler.py

```diff
@@ -57,16 +57,26 @@ def create_sb_output_dir(output_dir=None):
 def get_sb_config(config_file):
     """Read SuperBench config yaml.
 
-    Read config file, use default config if None is provided.
+    Read config file, detect Azure SKU and use corresponding config if None is provided.
 
     Args:
         config_file (str): config file path.
 
     Returns:
         OmegaConf: Config object, None if file does not exist.
     """
-    default_config_file = Path(__file__).parent / '../../config/default.yaml'
-    p = Path(config_file) if config_file else default_config_file
+    p = Path(str(config_file))
+    if not config_file:
+        config_path = Path(__file__).parent / '../../config'
+        p = config_path / 'default.yaml'
+        vm_size = get_vm_size().replace('Standard_', '').lower()
+        if vm_size:
+            logger.info('Detected Azure SKU standard_%s.', vm_size)
+            for config in (config_path / 'azure').glob('**/*'):
+                if config.name.startswith(vm_size):
+                    p = config
+                    break
+        logger.info('No benchmark config provided, using config file %s.', str(p))
```

Comment:
Maybe we cannot find the right configuration. Shall we output the error?

Reply:
we can only cover a few common SKUs, so it's just in best effort
if the SKU is detected but there's no config for it, the following info log will be shown instead of error:
```
[INFO] Detected Azure SKU ...
[INFO] ... using config file default.yaml
```

Reply:
Really confused here. Do we support default.yaml? It's strange for SuperBench has default.yaml. User should specify the yaml, otherwise, it can do auto-detection. If there is no support, we should print error

Reply:
> Do we support default.yaml?

yes

here's how the config is decided:
```
if config is specified:
    use provided config
else
    use default.yaml
    if sku can be detected and it happens to have a corresponding config:
        use sku.yaml
```
this PR only adds the "sku can be detected" feature, it will try to find a sku.yaml but no guarantee, for other cases (not on azure, unknow sku, etc.) default.yaml is stilled used as what previous code did
if we decide to remove default.yaml, it will be a breaking change and makes --config argument required instead of optional